### DISCUSSION TITLE
Unable to publish JAR ot Github Package Registry due to 404

### DISCUSSION THREAD

**Discussion by mridang** [community/community#26831] [Packages]
<p>I’m part of an organization and I’ve created a token for myself. I’ve updated my Gradle file to have a publication section.</p>
<pre><code>publishing{  
repositories{  
maven{  
name ="GitHubPackages"  
url = uri("https://maven.pkg.github.com/myorg/mypackage")
credentials{  
username ="mridang"  
password ="\*\*\*\*\*\*\*"  
}  
 }  
 }  
publications{  
gpr(MavenPublication){  
from(components.java)
}  
 }  
}
</code></pre>
<p>When I run <code>gradle publish</code> I see it being uploaded and then it fails with  </p>
<pre><code>Could not PUT 'https://maven.pkg.github.com/myorg/mypackage/com/myorg/mypackage/0.0.1-SNAPSHOT/mypackage-0.0.1-20200511.185059-1.jar'. Received status code 404 from server: Not Found
</code></pre>
<p>The group-id is com.myorg and I cannot seem to understand why it won’t publish.</p>

**Comment by jcansdale** [comment, marked as answer]
<p>I believe the problem is that you need to specify the repository you want to associate with your package, not the package name.</p>
<p>For example:</p>
<pre><code class="lang-auto">url = uri("https://maven.pkg.github.com/ORG/REPO")
</code></pre>
<p>The 404 is because it can’t find a repository called <code>mypackage </code>.</p>

**Comment by sw-dev2** [comment]
<p>Not working for an android java library. Private REPO exists, but  uri(“<a href="https://maven.pkg.github.com/ORG/REPO" rel="noopener nofollow ugc">https://maven.pkg.github.com/ORG/REPO</a>”) fails persistently with 404.</p>

**Comment by benrhine** [comment]
Try putting your raw user/token in. I messed with this forever and it ended up being that for some reason the publish task was not actually receiving the user/token
